builder/common/step_security_group.go

```diff
@@ -136,10 +137,39 @@ func (s *StepSecurityGroup) Run(ctx context.Context, state multistep.StateBag) m
 
 	log.Printf("[DEBUG] Found security group %s", s.createdGroupId)
 
+	temporarySGSourceCidrs := s.TemporarySGSourceCidrs
+	if len(temporarySGSourceCidrs) == 0 && s.TemporarySGSourcePublicIp {
```

Comment:
I'd suggest warning the user that if both arguments are used together, one will be ignored. This will prevent a source of confusion for users.

```suggestion
	if len(temporarySGSourceCidrs) != 0 && s.TemporarySGSourcePublicIp {
		ui.Warn("Using temporary_security_group_source_public_ip with temporary_security_group_source_cidrs is unsupported, ignoring temporary_security_group_source_public_ip.")
	}
	if len(temporarySGSourceCidrs) == 0 && s.TemporarySGSourcePublicIp {
```

I may be overcareful though, the documentation is pretty clear on what is expected, still I think this will help prevent surprises if we clearly state what the problem with the configuration is.

Reply:
Sure. Will add.

Reply:
done